src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UpdateRepresentationService.java

```diff
@@ -120,18 +112,17 @@ private Map<String, Object> updateCaseDataWithNewSolDetails(CaseDetails caseDeta
         String appSolicitorAddressField = isConsented ? CONSENTED_SOLICITOR_ADDRESS : CONTESTED_SOLICITOR_ADDRESS;
         String solicitorAddressField = isApplicant ? appSolicitorAddressField : RESP_SOLICITOR_ADDRESS;
 
-        //isRepresented Boolean
         caseData.put(isApplicant ? APPLICANT_REPRESENTED : getRespondentRepresentedKey(caseDetails), YES_VALUE);
         OrganisationsResponse organisationsResponse = organisationService
             .findOrganisationByOrgId(addedSolicitor.getOrganisation().getOrganisationID());
 
-        //address
         caseData.put(solicitorAddressField,
             updateSolicitorDetailsService.convertOrganisationAddressToSolicitorAddress(organisationsResponse));
 
-        //Other contact fields
-        return updateSolicitorDetailsService.updateSolicitorContactDetails(addedSolicitor, caseData,
-            isConsented, isApplicant);
+        Map<String, Object> updatedCaseData = updateSolicitorDetailsService.updateSolicitorContactDetails(addedSolicitor,
+             caseData, isConsented, isApplicant);
+
+        return updateSolicitorDetailsService.removeSolicitorFields(updatedCaseData, isConsented, isApplicant);
```

Comment:
Same here, chaining method calls to the return of your method is not great specially when both methods are transforming input objects